Client/Frontend/Shields/ShareTrackersController.swift

```diff
@@ -0,0 +1,338 @@
+    }
+    
+    // MARK: Properties
+    
+    private let theme: Theme
+    private let trackingType: TrackingType
+    
+    private let shareTrackersView: ShareTrackersView
+    
+    private lazy var gradientView = GradientView(
+        colors: [#colorLiteral(red: 0.968627451, green: 0.2274509804, blue: 0.1098039216, alpha: 1), #colorLiteral(red: 0.7490196078, green: 0.07843137255, blue: 0.6352941176, alpha: 1)],
+        positions: [0, 1],
+        startPoint: .zero,
+        endPoint: CGPoint(x: 1, y: 0.5))
+    
+    var actionHandler: ((Action) -> Void)?
+
+    // MARK: Lifecycle
+    
+    init(theme: Theme, trackingType: TrackingType) {
+        self.theme = theme
+        self.trackingType = trackingType
+        shareTrackersView = ShareTrackersView(trackingType: trackingType)
+        
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    @available(*, unavailable)
+    required init(coder: NSCoder) {
+        fatalError()
+    }
+    
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        
+        if #available(iOS 13.0, *) {
+            if traitCollection.userInterfaceStyle != previousTraitCollection?.userInterfaceStyle {
+                applyTheme(Theme.of(nil))
+            }
+        }
+    }
+    
+    // MARK: Internal
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        shareTrackersView.actionHandler = { [weak self] action in
+            switch action {
+            case .didShareWithMailTapped:
+                self?.actionHandler?(.takeALookTapped)
+            case .didShareWithTwitterTapped:
+                self?.actionHandler?(.shareTwitterTapped)
+            case .didShareWithFacebookTapped:
+                self?.actionHandler?(.shareFacebookTapped)
+            case .didShareWithDefaultTapped:
+                self?.actionHandler?(.shareMoreTapped)
+            case .didTakeALookTapped:
+                self?.actionHandler?(.takeALookTapped)
+            case .didDontShowTapped:
+                self?.actionHandler?(.dontShowAgainTapped)
+            }
+        }
+        
+        applyTheme(theme)
+        doLayout()
+    }
+    
+    private func doLayout() {
+        view.addSubview(shareTrackersView)
+
+        view.snp.makeConstraints {
+            $0.width.equalTo(264)
+            $0.height.equalTo(shareTrackersView)
+        }
+        
+        shareTrackersView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        
+        if case .trackerAdWarning = trackingType {
+            shareTrackersView.insertSubview(gradientView, at: 0)
+            
+            gradientView.snp.makeConstraints { make in
+                make.edges.equalToSuperview()
+            }
+        }
+    }
+    
+    // MARK: Themeable
+    
+    func applyTheme(_ theme: Theme) {
+        view.backgroundColor = UIColor(rgb: 0x339AF0)
```

Comment:
Colour is for this screen only? Maybe we make it a constant somewhere?

Reply:
Similar to other one a sad lonely color not used anywhere else